pkg/observer.go

```diff
@@ -0,0 +1,83 @@
+package ocr2keepers
+
+import "context"
+
+type UpkeepPayload interface{}
+type CheckResult interface{}
+
+type Tick interface {
+	// GetUpkeeps provides an array of upkeeps scoped to the individual tick
+	GetUpkeeps(ctx context.Context) ([]UpkeepPayload, error)
+}
+
+// Runner2 is the interface for an object that should determine eligibility state
+// (I didn't want to mess with the existing runner and was unsure if we wanted to reuse that)
```

Comment:
Good callout. We do want to re-use that and there is a ticket for it as well. I'll add a note in that ticket to unify them both, but first a new function will need to be added to the existing runner and the function signature updated.